spec/decorators/notification_decorator_spec.rb

```diff
@@ -0,0 +1,63 @@
+require "rails_helper"
+
+RSpec.describe NotificationDecorator, type: :decorator do
+  describe "#mocked_object" do
+    let(:comment) { create(:comment, commentable: create(:article, organization: create(:organization))) }
+
+    it "returns empty struct if the notification is new" do
+      notification = build(:notification)
+      result = notification.decorate.mocked_object("user")
+
+      expect(result.name).to be_empty
+      expect(result.id).to be_nil
+    end
+
+    it "returns empty struct class and its name if the notification is new" do
+      notification = build(:notification)
+      result = notification.decorate.mocked_object("user")
+
+      expect(result.class).to be_a(Struct)
+      expect(result.class.name).to be_empty
+    end
+
+    it "returns class name and id for the reactable in a struct" do
+      notification = Notification.send_new_comment_notifications_without_delay(comment)
+
+      result = notification.decorate.mocked_object("user")
+      expect(result.name).to eq("User")
+      expect(result.id).to eq(comment.user.id)
+    end
+
+    it "returns struct class and its name" do
+      notification = Notification.send_new_comment_notifications_without_delay(comment)
```

Comment:
might be nice to pull this out into a `let` since we're using it in a few places :) but just a nice to have, not a requirement.

Reply:
good call, i'll write it down for a following PR!